src/trace/dma-trace.c

```diff
@@ -475,26 +475,34 @@ void dma_trace_flush(void *t)
 void dma_trace_on(void)
 {
 	struct dma_trace_data *trace_data = dma_trace_data_get();
+	uint32_t flags;
 
 	if (!trace_data || trace_data->enabled)
 		return;
 
-	trace_data->enabled = 1;
+	spin_lock_irq(&trace_data->lock, flags);
+
 	schedule_task(&trace_data->dmat_work, DMA_TRACE_PERIOD,
 		      DMA_TRACE_PERIOD);
+	trace_data->enabled = 1;
```

Comment:
you're setting this flag protected by the spinlock, but you test it on like 480 without any protection.
Is this intentional?